src/extensions/data_array.js

```diff
@@ -67,24 +96,101 @@ export default () => {
       });
     };
 
+    const createInstances = observationObjects => {
+      return new Promise(resolve => {
+        db().then(models => {
+          Promise.all(observationObjects.map(observationObj => {
+            const newReq = Object.create(req);
+            newReq.body = observationObj.body;
+            return models.createInstance(observations, newReq, exclude).then(
+              instance => {
+              observationObj.instanceId = instance.id;
+              return Promise.resolve();
+            }).catch(() => {
+              return Promise.resolve();
+            });
+          })).then(results => {
+            return resolve(results);
+          });
+        }).catch(() => {
+          return ERR.ApiError(res, 500,
+            ERR.ERRNO_INTERNAL_ERROR, ERR.INTERNAL_ERROR);
+        });
+      });
+    };
+
+    const createObservations = body => {
+      let observationObjects = [];
+      const resultIndex = body.components.indexOf(result);
+      const featureOfInterestIndex =
+        body.components.indexOf(featureOfInterest + '/id');
+      const phenomenonTimeIndex = body.components.indexOf(phenomenonTime);
+      const dataArrayCount = body[dataArrayIotCount];
+
+      for (let i = 0; i < dataArrayCount; i++) {
+        let observation = {};
+        observation.Datastream = {};
+        observation.Datastream[iotId] = body.Datastream[iotId];
```

Comment:
I cannot see the code where we check the existence of the Datastream. Are we delegating this to sequelize enforcing its foreign key constraints? If that is the case, we should add a comment about this somewhere in the code.